KenticoCloud.Delivery/StrongTyping/DeliveryItemsFeed.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace KenticoCloud.Delivery
+{
+    /// <summary>
+    /// Represents a feed that can be used to retrieve strongly typed content items from Kentico Cloud Delivery API in smaller batches.
+    /// </summary>
+    /// <typeparam name="T">The type of content items in the feed.</typeparam>
+    public class DeliveryItemsFeed<T> : IEnumerable<T>
+    {
+        internal delegate Task<DeliveryItemsFeedResponse<T>> GetFeedResponse(string continuationToken);
+
+        private DeliveryItemsFeedResponse<T> _lastResponse;
+        private readonly GetFeedResponse _getFeedResponseAsync;
+
+        /// <summary>
+        /// Indicates whether there are more batches to fetch.
+        /// </summary>
+        public bool HasMoreResults => _lastResponse == null || !string.IsNullOrEmpty(_lastResponse.ContinuationToken);
```

Comment:
@sutrkiller HasMoreResults is true if last response is null, but that doesn't necessarily means there are more results. I would consider adding new property to indicate if some batch was already fetched and 'HasMoreResults' should be equal to true only if there is valid continuationToken

Reply:
I agree that the name 'HasMoreResult' might be misleading but a call to FetchNextBatch would be necessary anyway. Again, conveniance was a main reason for this I cannot come up with any use case for the new proposed property as the expected usage of Feed is something like:
```
while (feed.HasMoreResults) 
{
    var nextBatch = await feed.FetchNextBatchAsync();
   // do something with items
}
```
or do-while alternatively...

Reply:
Yeah, I guess the example you posted is reasonable and easy to use so we can leave this as it is.